instrumentation/opentelemetry-extension-annotations-1.0/javaagent/src/test/java/io/opentelemetry/test/annotation/WithSpanInstrumentationTest.java

```diff
@@ -35,225 +36,155 @@ class WithSpanInstrumentationTest {
-                                                    CodeIncubatingAttributes.CODE_NAMESPACE,
-                                                    TracedWithSpan.class.getName()),
-                                                entry(
-                                                    CodeIncubatingAttributes.CODE_FUNCTION,
-                                                    "completionStage")))));
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("TracedWithSpan.completionStage")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasNoParent()
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(
+                                CodeIncubatingAttributes.CODE_NAMESPACE,
+                                TracedWithSpan.class.getName()),
+                            equalTo(CodeIncubatingAttributes.CODE_FUNCTION, "completionStage"))));
   }
 
   @Test
-  void exceptionallyCompletedCompletionStage() throws Exception {
-
+  void exceptionallyCompletedCompletionStage() {
     CompletableFuture<String> future = new CompletableFuture<>();
     future.completeExceptionally(new IllegalArgumentException("Boom"));
     new TracedWithSpan().completionStage(future);
 
-    assertThat(testing.waitForTraces(1))
-        .satisfiesExactly(
-            trace ->
-                assertThat(trace)
-                    .satisfiesExactly(
-                        span ->
-                            assertThat(span)
-                                .hasName("TracedWithSpan.completionStage")
-                                .hasKind(SpanKind.INTERNAL)
-                                .hasParentSpanId(SpanId.getInvalid())
-                                .hasStatus(StatusData.error())
-                                .hasAttributesSatisfying(
-                                    attributes ->
-                                        assertThat(attributes)
-                                            .containsOnly(
-                                                entry(
-                                                    CodeIncubatingAttributes.CODE_NAMESPACE,
-                                                    TracedWithSpan.class.getName()),
-                                                entry(
-                                                    CodeIncubatingAttributes.CODE_FUNCTION,
-                                                    "completionStage")))));
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("TracedWithSpan.completionStage")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasNoParent()
+                        .hasStatus(StatusData.error())
+                        .hasException(new IllegalArgumentException("Boom"))
```

Comment:
[question] I haven't checked how the exception check is being implemented, wouldn't it be easier to reuse the same `IllegalArgumentException` instance ? Because those two instances do not have the exact same stack trace by being distinct instances.

Reply:
[hasException](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/SpanDataAssert.java#L408-L411) checks the message and type contents, but for the stack trace it just checks that it is not null.

I updated the tests to use and assert the same exception object